### DISCUSSION TITLE
Failure to run tests in parallel

### DISCUSSION THREAD

**Discussion by DanyalKhawaja** [NomicFoundation/hardhat#3093] [Troubleshooting]
I am having the following errors when running testing contracts with the --parallel option. It actually tries to re-acquire the lock on openzappelin manifest file and fails. I have no clue at all what to do!

![image](https://user-images.githubusercontent.com/42298472/187088395-530ec0b1-53db-4fab-b35c-c15a60abbb2b.png)
![image](https://user-images.githubusercontent.com/42298472/187088440-615a2fb1-4caf-44da-89a4-545485d27689.png)

**Comment by frangio** [comment]
Hi @DanyalKhawaja, we recommend avoiding `--parallel` for now when using OpenZeppelin Upgrades Plugins until we implement better handling for the specifics of parallel tests. https://github.com/OpenZeppelin/openzeppelin-upgrades/issues/645
